### PR TITLE
Add a class that makes testing MetricsCollectors easier

### PR DESCRIPTION
**Pull request by sidheart**
*Fixes #:* 
N/A

*Description of changes:*
The AbstractCollectorTest class can be extended to validate the behavior
of any PerformanceAnalyzerMetricsCollector.

*Tests:*
- AbstractCollectorTest
- GCInfoCollectorTest

*If new tests are added, how long do the new ones take to complete*
- Will be based on the sum of all collectMetrics() calls, which should not exceed tens of seconds.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  // Implementors should call this in a setup() function to set their collector
+  public void setUut(PerformanceAnalyzerMetricsCollector collector) {
+    this.uut = collector;
+  }
+
+  // This is the only line that implementors need to modify
+  public abstract void validateMetric(String metric) throws Exception;
+
+  // This function does not need to be modified
```
Comment (yojs) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java:21]: Is this comment relevant ?

**Review by yojs — APPROVED** [review]

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  @Test
+  public void validateMetrics() throws Exception {
+    uut.collectMetrics(Instant.now().toEpochMilli());
+    String metricString = uut.getValue().toString();
+    // chop off current time json
+    int end = metricString.indexOf(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+    metricString = metricString.substring(end+1);
+    while (!metricString.isEmpty()) {
+      end = metricString.indexOf(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+      if (end < 0) {
```
Comment (ktkrg) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java:31]: why not `end == -1` ?

**Review by ktkrg — COMMENTED** [review]
Looks like checkstyle is failing.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  // This is the only line that implementors need to modify
+  public abstract void validateMetric(String metric) throws Exception;
+
+  // This function does not need to be modified
+  @Test
+  public void validateMetrics() throws Exception {
+    uut.collectMetrics(Instant.now().toEpochMilli());
+    String metricString = uut.getValue().toString();
```
Comment (aditjind) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java:25]: I may not be understanding something correctly here, but don't we need an implementation of getValue in every collector? I was unable to locate one in the `GCInfoCollector`

**Pull request merged**